server/src/internalClusterTest/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java

```diff
@@ -436,6 +437,43 @@ public void testRestoreSnapshotAllocationDoesNotExceedWatermark() throws Excepti
         assertBusyWithDiskUsageRefresh(dataNode0Id, indexName, hasSize(1));
     }
 
+    public void testDiskMonitorResetLastRuntimeMilliSecOnlyInFirstCall() throws Exception {
+        final Settings settings = Settings.builder()
+            .put(DiskThresholdSettings.CLUSTER_CREATE_INDEX_BLOCK_AUTO_RELEASE.getKey(), false)
+            .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_DISK_THRESHOLD_ENABLED_SETTING.getKey(), false)
+            .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_REROUTE_INTERVAL_SETTING.getKey(), TimeValue.timeValueSeconds(5))
+            .build();
+
+        internalCluster().startClusterManagerOnlyNode(settings);
+        internalCluster().startDataOnlyNodes(2, settings);
+        ensureStableCluster(3);
+
+        final MockInternalClusterInfoService clusterInfoService = getMockInternalClusterInfoService();
+        // Reduce disk space of all node.
+        clusterInfoService.setDiskUsageFunctionAndRefresh((discoveryNode, fsInfoPath) -> setDiskUsage(fsInfoPath, TOTAL_SPACE_BYTES, 0));
+
+        Thread.sleep(TimeValue.timeValueSeconds(5).getMillis());
+
+        // DiskMonitor reevaluates block at 0 space.
+        clusterInfoService.setDiskUsageFunctionAndRefresh((discoveryNode, fsInfoPath) -> setDiskUsage(fsInfoPath, TOTAL_SPACE_BYTES, 0));
+
+        Thread.sleep(300);
+
+        // User removes index create block.
+        Settings removeBlockSetting = Settings.builder().put(Metadata.SETTING_CREATE_INDEX_BLOCK_SETTING.getKey(), "false").build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(removeBlockSetting).get());
+        // Free all the space
+        clusterInfoService.setDiskUsageFunctionAndRefresh(
+            (discoveryNode, fsInfoPath) -> setDiskUsage(fsInfoPath, TOTAL_SPACE_BYTES, TOTAL_SPACE_BYTES)
+        );
+
+        // Sleep for 30 sec to update blocks metadata.
+        Thread.sleep(30000);
```

Comment:
too long for tests to get blocked on completion.

Reply:
@RS146BIJAY Can you get the coverage you want with a fast and deterministic unit test?

Reply:
Actually this is for metadata update. It can take 30 sec for the block to get [reflected](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/internalClusterTest/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java#L324). I will replace this with assertBusy as used in other test case.

Reply:
Addressed the comment.

Reply:
Checking what is causing the test cases to be flaky

Reply:
@andrross I don't this change can be tested with unit tests. Issue is [LastRuntime is set after a response is returned by index create block call](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdMonitor.java#L425). Will be difficult to mock this scenario. I have made test case deterministic.

Reply:
Can we override the monitor run interval in tests?